common/docker/client.go

```diff
@@ -72,21 +78,29 @@ func (c *Client) Info() string {
 
 // AddObserver adds an observer for docker events
 func (c *Client) AddObserver(ob ContainerObserver) error {
-	events := make(chan *docker.APIEvents)
-	if err := c.AddEventListener(events); err != nil {
-		Log.Errorf("[docker] Unable to add listener to Docker API: %s", err)
-		return err
-	}
-
 	go func() {
-		for event := range events {
-			switch event.Status {
-			case "start":
-				id := event.ID
-				ob.ContainerStarted(id)
-			case "die":
-				id := event.ID
-				ob.ContainerDied(id)
+		retryInterval := InitialInterval
```

Comment:
It's not bad. Consider if we fail to connect a number of times, and the retry interval goes up to  1 minute.  Then we connect, but it fails again after 30 seconds.  We will now wait 1 minute before trying again.

However, having had various long pauses while we retried, this doesn't seem much more terrible.

I guess the back-off is just to stop killing the machine with rapid reconnects.